### PR TITLE
add grpc compliant health check

### PR DESCRIPTION
**Pull request by kwkr**
Added health check implementation according to #2614 
I added only the method `Check` that always returns `Serving`. Currently the other method `Watch` is omitted as it looks like it would need some additional packages `tokio_stream`. If it needs some further changes, please let me know and I can take care of it.

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by timvisee** [comment]
Thank you for implementing this. I thought this health check had to replace our current health checking mechanism which would be breaking, it turns out it's just a simple extra message. I'll try to properly review this tomorrow.

**Review by timvisee — APPROVED** [review]
Thank you for implementing this!

File: lib/api/src/grpc/proto/health_check.proto [hunk 1]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
```
Comment (agourlay) [inline lib/api/src/grpc/proto/health_check.proto:1]: I'd love a small comment pointing to the reference https://github.com/grpc/grpc/blob/master/doc/health-checking.md#service-definition

**Review by agourlay — APPROVED** [review]
Thanks for taking care of that :+1:

File: lib/api/src/grpc/proto/health_check.proto (hunk 1 above)
Comment (kwkr) [reply lib/api/src/grpc/proto/health_check.proto:2]: Added the link as a comment.

**Review by generall — APPROVED** [review]

**Pull request merged**